main.go

```diff
@@ -40,15 +46,17 @@ func Start(ctx context.Context, version string) (*Server, error) {
 	}
 
 	// Create a db dir. Even the ephemeralForTest engine needs a dbpath.
-	server.dbDir, err = ioutil.TempDir("", "")
+	server.dbDir, err = os.MkdirTemp("", "")
 	if err != nil {
 		log.Fatal(ctx, "Error creating data directory", err)
 		return nil, err
 	}
 
 	log.Info(ctx, "Starting mongod server", log.Data{"binPath": binPath, "dbDir": server.dbDir})
-	// By specifying port 0, mongo will find and use an available port
-	server.cmd = exec.Command(binPath, "--storageEngine", "ephemeralForTest", "--dbpath", server.dbDir, "--port", "0")
+
+	// Find a free port for the server - unfortunately the initial idea of allowing the mongo server to chose its own port
+	// (by setting a port of 0 in the server commandline) will not work as it interferes with the later replica set initialisation
+	server.cmd = exec.Command(binPath, "--replSet", "rs0", "--dbpath", server.dbDir, "--port", getFreeMongoPort())
```

Comment:
Should we use `server.ReplicaSet()`?

```suggestion
	server.cmd = exec.Command(binPath, "--replSet", server.ReplicaSet(), "--dbpath", server.dbDir, "--port", getFreeMongoPort())
```